lib/PrimitiveHelpers.js

```diff
@@ -0,0 +1,213 @@
+'use strict';
+var Cesium = require('cesium');
+var deepEqual = require('deep-equal');
+var AccessorReader = require('./AccessorReader');
+var readAccessor = require('./readAccessor');
+var writeAccessor = require('./writeAccessor');
+
+var Cartesian3 = Cesium.Cartesian3;
+var Matrix4 = Cesium.Matrix4;
+var defined = Cesium.defined;
+
+module.exports = {
+    getAllPrimitives : getAllPrimitives,
+    getPrimitivesByMaterialMode : getPrimitivesByMaterialMode,
+    markPrimitiveConflicts : markPrimitiveConflicts,
+    primitiveEquals : primitiveEquals,
+    primitivesShareAttributeAccessor : primitivesShareAttributeAccessor,
+    primitivesHaveOverlappingIndexAccessors : primitivesHaveOverlappingIndexAccessors,
+    transformPrimitives : transformPrimitives
+};
+
+function primitivesShareAttributeAccessor(primitive, comparePrimitive) {
+    var attributes = primitive.attributes;
+    var compareAttributes = comparePrimitive.attributes;
+    for (var attribute in attributes) {
+        if (attributes.hasOwnProperty(attribute)) {
+            if (compareAttributes.hasOwnProperty(attribute)) {
+                if (attributes[attribute] === compareAttributes[attribute]) {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+function primitivesHaveOverlappingIndexAccessors(gltf, primitive, comparePrimitive) {
+    var accessors = gltf.accessors;
+    var indexAccessorId = primitive.indices;
+    var compareIndexAccessorId = comparePrimitive.indices;
+    if (!defined(indexAccessorId) || !defined(compareIndexAccessorId)) {
+        return false;
+    }
+    if (indexAccessorId === compareIndexAccessorId) {
+        return true;
+    }
+    var indexAccessor = accessors[indexAccessorId];
+    var compareIndexAccessor = accessors[compareIndexAccessorId];
+    var indices = [];
+    readAccessor(gltf, indexAccessor, indices);
+    var accessorReader = new AccessorReader(gltf, compareIndexAccessor);
+    var value = [];
+
+    while(accessorReader.hasNext()) {
+        var index = accessorReader.read(value)[0];
+        if (indices.indexOf(index) >= 0) {
+            return true;
+        }
+        accessorReader.next();
+    }
+    return false;
+}
+
+function transformPrimitives(gltf, primitives, transform) {
+    var inverseTranspose = new Matrix4();
```

Comment:
However, this should use `AccessorReader` instead of `readAccessor`. I will make that change.